java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -254,6 +281,26 @@ public interface PreparedStatement extends AutoCloseable {
    * @return a new prepared statement.
    */
   public PreparedStatement prepare(final String query) {
+    if (setCatalogInSession) {
```

Comment:
Is there some way to do this at initial connection time? (Some things may not prepare a statement but we'd still want the catalog/session to exist right? e.g. if someone connects and queries the catalog/metadata)

Reply:
Yes, valid point. I think it makes sense to move this to where the connection is created as that is more appropriate.